Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM golang:1.17.3
+
+ADD . cloud-api-adaptor
+RUN git clone -b ccv0-peer-pods https://github.com/bpradipt/kata-containers
+RUN apt-get update && apt-get install -y libvirt-dev libvirt0 genisoimage # other requirements?
+WORKDIR cloud-api-adaptor
+# generic build
+ENV CLOUD_PROVIDER=aws
+RUN make
+RUN install cloud-api-adaptor /usr/local/bin/cloud-api-adaptor
+RUN make clean
+# libvirt build
+ENV CLOUD_PROVIDER=libvirt
+RUN make
+RUN install cloud-api-adaptor /usr/local/bin/cloud-api-adaptor-libvirt
```

Comment:
For libvirt we don't have a static build yet so we need the dependencies (libvirt-dev etc). However for the cloud providers we have a static build. So we can use a distroless base image (eg `gcr.io/distroless/static`) for the actual binary.

Reply:
I see, i can do so and also have a makefile target for each provider as you suggested but it will require us to maintain 2 container images (and different yamls?) instead of a single container which contains both binaries.
Is it preferable? WDYT?

Reply:
Let's avoid maintaining multiple container images for now and stick to only one.